jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt

```diff
@@ -61,73 +25,19 @@ class JavaLambdaBuilder : LambdaBuilder() {
         envVars: Map<String, String>,
         samOptions: SamOptions
     ): CompletionStage<BuiltLambda> {
-        val buildDir = FileUtil.createTempDirectory("lambdaBuild", null, true)
-
-        val future = CompletableFuture<BuiltLambda>()
-        val compilerManager = CompilerManager.getInstance(module.project)
-        val compileScope = compilerManager.createModulesCompileScope(arrayOf(module), true, true)
+        val baseDir = getBaseDirectory(module).path
+        val customTemplate = FileUtil.createTempFile("template", ".yaml", true)
+        SamTemplateUtils.writeDummySamTemplate(customTemplate, runtime, baseDir, handler, envVars)
 
-        compilerManager.make(compileScope) { aborted, errors, _, context ->
-            if (!aborted && errors == 0) {
-                try {
-                    copyLibraries(module, buildDir)
-                    copyClasses(module, buildDir)
-                    future.complete(BuiltLambda(null, buildDir.toPath()))
-                } catch (e: Exception) {
-                    future.completeExceptionally(RuntimeException(message("lambda.package.zip_fail"), e))
-                }
-            } else if (aborted) {
-                future.completeExceptionally(RuntimeException(message("lambda.package.compilation_aborted")))
-            } else {
-                val errorMessages = context.getMessages(CompilerMessageCategory.ERROR).joinToString("\n")
-                future.completeExceptionally(
-                    RuntimeException(
-                        message(
-                            "lambda.package.compilation_errors",
-                            errorMessages
-                        )
-                    )
-                )
-            }
-        }
-        return future
-    }
-
-    private fun copyLibraries(module: Module, buildDir: File) {
-        val libDir = File(buildDir, "lib")
-        productionRuntimeEntries(module)
-            .librariesOnly()
-            .pathsList.pathList
-            .map { File(it) }
-            .filter { it.exists() }
-            .forEach { copyFileOrDir(it, libDir) }
-    }
-
-    private fun copyClasses(module: Module, buildDir: File) {
-        productionRuntimeEntries(module)
-            .withoutLibraries()
-            .classes()
-            .pathsList.pathList
-            .map { File(it) }
-            .filter { it.exists() }
-            .forEach { copyFileOrDir(it, buildDir) }
+        return buildLambdaFromTemplate(module, customTemplate.toPath(), "Function", samOptions)
```

Comment:
Do we need to return this magical logical id `Function` from `writeDummySamTemplate` method call? It seems a bit inexplicable here.

Reply:
We should probably just pass it in, its hard coded right now